testcases/open_posix_testsuite/conformance/interfaces/feof/2-1.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is licensed under the GPL license.  For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *
+ *	 If the stream is valid, feof function shall not change the setting of errno.
+ *
+ * method:
+ *	-open file in write mode
+ *	-write a sample string into the file
+ *	-close the file
+ *	-open the same file in read mode so that a valid stream is obtained
+ *	-check feof return value and errno
+ *	-read the  string from the file using fscanf
+ *	-check feof return value and errno
+ */
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "posixtest.h"
+
+#define TNAME "feof/2-1.c"
+#define FNAME "in_file"
+#define STRING_CONST "LINUX"
+
+static void remove_file(FILE *in_fp)
+{
+	errno = 0;
+	if (in_fp != NULL) {
+		if (fclose(in_fp) == EOF) {
+			printf(TNAME " Error at fclose(), errno = %d\n", errno);
+			unlink(FNAME);
+			exit(PTS_UNRESOLVED);
+		}
+	}
+	unlink(FNAME);
+}
+
+int main(void)
+{
+	FILE *in_fp;
+	char sample_string[strlen(STRING_CONST)];
+	int ret;
+	errno = 0;
+
+	in_fp = fopen(FNAME, "w");
+	if (in_fp == NULL) {
+		printf(TNAME " Error at fopen(), errno = %d\n", errno);
+		exit(PTS_UNRESOLVED);
+	}
+
+	ret = fprintf(in_fp, "%s", STRING_CONST);
+	if (ret < 0) {
+		printf(TNAME " Error at fprintf()\n");
+		remove_file(in_fp);
+		exit(PTS_UNRESOLVED);
+	}
+
+	errno = 0;
+	if (fclose(in_fp) == EOF) {
+		printf(TNAME " Error at fclose(), errno = %d\n", errno);
+		unlink(FNAME);
+		exit(PTS_UNRESOLVED);
+	}
+
+	errno = 0;
+	in_fp = fopen(FNAME, "r");
+	if (in_fp == NULL) {
+		printf(TNAME " Error at fopen(), errno = %d\n", errno);
+		remove_file(in_fp);
+		exit(PTS_UNRESOLVED);
+	}
+
+	errno = 0;
+	/* If stream is valid, errno should not be set by feof */
+	if (!feof(in_fp) && errno == 0) {
+		ret = fscanf(in_fp, "%s", sample_string);
+		if (ret != 1 && errno != 0) {
+			printf(TNAME " Unexpected error at fscanf, errno = %d\n", errno);
+			remove_file(in_fp);
+			exit(PTS_UNRESOLVED);
+		} else {
+			/* If stream is valid, errno should not be set by feof */
+			if (feof(in_fp) && errno == 0) {
+				printf(TNAME " Test Passed.\n");
+				remove_file(in_fp);
+				exit(PTS_PASS);
+			} else {
+				printf(TNAME " Test Failed.\n");
+				remove_file(in_fp);
+				exit(PTS_FAIL);
+			}
+		}
+	} else {
+		printf(TNAME " Test Failed.\n");
+		remove_file(in_fp);
+		exit(PTS_FAIL);
+	}
+}
```

Comment:
This is mostly the same as the previous test, partly copy&pasted, hence the same applies to this one as well.

Reply:
I think it is best to have some other logical value for errno. 0 is an unused value(undefined).

Reply:
Some implementations of feof(3) may set the value of `errno` to 0 though. I think another bogus value would be better here, as long as it's not too esoteric.